main/autogen/subclasses.cc

```diff
@@ -140,30 +173,31 @@ vector<string> Subclasses::serializeSubclassMap(const Subclasses::Map &descendan
 //
 // This effectively replaces pay-server's `DescendantsMap` in `InheritedClassesStep` with a much
 // faster implementation.
-vector<string> Subclasses::genDescendantsMap(Subclasses::Map &childMap, vector<string> &parentNames) {
-    Subclasses::patchChildMap(childMap);
+vector<string> Subclasses::genDescendantsMap(const core::GlobalState &gs, Subclasses::Map &childMap,
+                                             vector<core::SymbolRef> &parentRefs, const bool showPaths) {
+    Subclasses::patchChildMap(gs, childMap);
 
     // Generate descendants for each passed-in superclass
-    fast_sort(parentNames);
+    fast_sort(parentRefs, [&gs](const auto &left, const auto &right) { return left.show(gs) < right.show(gs); });
     Subclasses::Map descendantsMap;
-    for (const string &parentName : parentNames) {
+    for (const auto &parentRef : parentRefs) {
         // Skip parents that the user asked for but which don't
         // exist or are never subclassed.
-        auto fnd = childMap.find(parentName);
-        if (fnd == childMap.end()) {
+        auto fnd = childMap.find(parentRef);
+        if (fnd == childMap.end() || !parentRef.exists()) {
```

Comment:
This `.exists()` looks a little suspect. I haven't traced the code thoroughly, but I'm guessing that we will want to make it so that if a parentRef doesn't exist, it's never even added to parentRefs in the first place.

Do you think we could maintain that invariant?

I'm _guessing_ that this is happening because your getConstantRef unconditionally inserts into the vector even if attempting to look up the hard-coded class didn't exist. If that's what's causing us to need this .exists here, I think we should only `emplace_back` into the parentRefs if getConstantRef actually found a suitable symbol.